woocommerce/woocommerce-ios
# Coupons: Add error state view to coupon list when coupons are disabled for a store

Closes: #6250


### Description

This PR updates the UI part for #6250:
- Add new error state and a corresponding overlay view for the coupon list when coupons are disabled for a store
- Integrate `retrieveCouponSetting` -  this is called in 2 cases:
  - when synchronizing the first page completes with no coupons found - we need to check if coupons are disabled
  - when enabling coupons fails - we need to check again to find the correct state 
- Integrate `enableCoupons` when a user taps on the action button of the error state.
- Bonus: Fix JetpackRequest for PUT method - we should treat PUT as POST for the funneled requests to succeed.

### Testing instructions

- Turn off coupons for your test store in WP Admin > WC Settings > General > Enable Coupons.
- Switch to the store if needed, navigate to Menu > Coupons. Notice that an error state view is displayed with an action button to enable coupons.
- Tap Enable Coupons, notice that after the request succeeds, you should be able to see your list of coupons again.

### Screenshots


https://user-images.githubusercontent.com/5533851/158727982-e0e7dc21-1b88-415d-85da-1d65e643b80f.mp4


---
